cli/tests/test_git_push.rs

```diff
@@ -238,6 +238,118 @@ fn test_git_push_not_fast_forward() {
     "###);
 }
 
+// Short-term TODO: implement this.
+// This tests whether the push checks that the remote branches are in expected
+// positions. Once this is implemented, `jj git push` will be similar to `git
+// push --force-with-lease`
+#[test]
+fn test_git_push_sideways_unexpectedly_moved() {
+    let (test_env, workspace_root) = set_up();
+
+    // Move branch1 forward on the remote
+    let origin_path = test_env.env_root().join("origin");
+    test_env.jj_cmd_ok(&origin_path, &["new", "branch1", "-m=remote"]);
+    std::fs::write(origin_path.join("remote"), "remote").unwrap();
+    test_env.jj_cmd_ok(&origin_path, &["branch", "set", "branch1"]);
+    let assert = test_env
+        .jj_cmd(&origin_path, &["branch", "list", "--all"])
+        .assert()
+        .success();
+    insta::assert_snapshot!(get_stdout_string(&assert), @r###"
```

Comment:
nit: `.jj_cmd_success()`, `_ok()`, or `_failure()` can be used instead of `.assert()`s.

I've also sent PR that adds `get_branch_output()` helper.
https://github.com/martinvonz/jj/pull/3715

Reply:
I was too lazy to fix this in all commits, but it should be fixed by the main "force-with-lease" commit.